WordPress/Classes/ViewRelated/Post/WPPostViewController.m

```diff
@@ -1411,6 +1411,10 @@ - (void)savePost
     [self logSavePostStats];
 
     [self.view endEditing:YES];
+
+    //Make sure that we are uploading the latest content on the editor.
+    self.post.postTitle = self.titleText;
+    self.post.content = self.bodyText;
```

Comment:
@aerych autoSaveContent is already being called when the image finished upload, and the race condition we see is between the save happening to CoreData and being reflected up to the self.post instance and the upload to the server.

The autosave is done in async way and I think that is the source of our problems. But we cannot change it to sync, because the method is being called for almost every character changed...